### PR TITLE
Add Spark remote debug model into SparkSubmitModel

### PR DESCRIPTION
**Pull request by djlan**
Add Spark remote debug model into SparkSubmitModel

### CODE REVIEW

**Comment by vscjavaci** [comment]
Can one of the admins verify this patch?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.microsoft.azure.hdinsight.spark.common;
+
+public class SparkSubmitAdvancedConfigModel {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: `SparkSubmitAdvancedDebugConfigModel` might be more suitable.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+    public boolean enableRemoteDebug = false;
+
+    public String sshUserName = "sshuser";
+
+    public SSHPasswordKeyOrLabel sshPasswordKeyOrLabel = SSHPasswordKeyOrLabel.UseArtifact;
+    public String sshKyeFile = "";
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: I need a `File` instance here

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    public String sshPassword = "";
+
+    public enum SSHPasswordKeyOrLabel {
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: `SSHAuthType` might be more suitable.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java [hunk 4]
```diff
@@ -66,6 +67,7 @@
     private Map<String, Artifact> artifactHashMap = new HashMap<>();
 
     private SparkSubmissionParameter submissionParameter;
+    private SparkSubmitAdvancedConfigModel advancedConfigModel;
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java]: `advanceDebugConfigModel`

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialog.java [hunk 5]
```diff
@@ -55,6 +55,7 @@
     public SparkSubmissionExDialog(@NotNull Project project, @NotNull List<IClusterDetail> cachedClusterDetails, @Nullable CallBack callBack) {
         this.project = project;
         this.callBack = callBack;
+
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialog.java]: can ignore this file change.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java (hunk 1 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: We might add more advanced config later so prefer to keep it general for now

---
In reply to: [117161075](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117161075) [](ancestors = 117161075)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java (hunk 2 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: Discuss offline.

---
In reply to: [117161118](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117161118) [](ancestors = 117161118)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java (hunk 3 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: Fixed

---
In reply to: [117161280](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117161280) [](ancestors = 117161280)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java (hunk 4 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitModel.java]: Same with above

---
In reply to: [117161326](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117161326) [](ancestors = 117161326)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java (hunk 2 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/common/SparkSubmitAdvancedConfigModel.java]: Fixed

---
In reply to: [117174510](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117174510) [](ancestors = 117174510,117161118)

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialog.java (hunk 5 above)
Comment (djlan) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionExDialog.java]: What does it mean?

---
In reply to: [117161406](https://github.com/Microsoft/azure-tools-for-java/pull/459#discussion_r117161406) [](ancestors = 117161406)

**Pull request merged**